coursera/utils.py

```diff
@@ -57,27 +57,39 @@ def random_string(length):
     return ''.join(random.choice(valid_chars) for i in range(length))
 
 
-def clean_filename(s, minimal_change=False):
+def clean_filename(s, minimal_change=False, keep_slashes=False):
     """
     Sanitize a string to be used as a filename.
 
-    If minimal_change is set to true, then we only strip the bare minimum of
+    If minimal_change is set to True, then we only strip the bare minimum of
     characters that are problematic for filesystems (namely, ':', '/' and
     '\x00', '\n').
+
+    If keep_slashes is set to True, we don't remove the '/' characters, as
+    we may be receiving a full filename.
     """
 
     # First, deal with URL encoded strings
-    h = html_parser.HTMLParser()
-    s = h.unescape(s)
+    # h = html_parser.HTMLParser()
+    s = unquote(s)
```

Comment:
I've partially addressed that too:
https://github.com/coursera-dl/coursera-dl/blob/master/coursera/utils.py#L85